.teamcity/step-scripts/calypso-unit-tests.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+set -o errexit
+set -o nounset
+set -o pipefail
+
+#### Run Type Checks
+export NODE_ENV="test"
+# TODO: parallelize these
+yarn tsc --build packages/*/tsconfig.json
+yarn tsc --build apps/editing-toolkit/tsconfig.json
+yarn tsc --build client/tsconfig.json
+yarn tsc --build test/e2e/tsconfig.json
+
+
+#### Run unit tests
+# TODO: parallelize these
+unset NODE_ENV
+unset CALYPSO_ENV
+yarn test-client --maxWorkers=$JEST_MAX_WORKERS --ci --reporters=default --reporters=jest-teamcity --silent
+yarn test-server --maxWorkers=$JEST_MAX_WORKERS --ci --reporters=default --reporters=jest-teamcity --silent
+yarn test-packages --maxWorkers=$JEST_MAX_WORKERS --ci --reporters=default --reporters=jest-teamcity --silent
+yarn test-build-tools --maxWorkers=$JEST_MAX_WORKERS --ci --reporters=default --reporters=jest-teamcity --silent
+yarn workspaces foreach --verbose --parallel run storybook --ci --smoke-test
```

Comment:
**thought:non-blockingI** - I have mixed thoughts about moving contents inside a script when that script won't be reused anywhere else. 

On one hand, it shortens the main Kotlin file, making it easier to navigate.
But to make changes a developer must follow the "cookie crumbs" to get to where the actual changes need to be made.

Reply:
I mainly did it since I needed to write more normal bash, so I wanted to have syntax highlighting and stuff